azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -61,14 +61,6 @@ class GoalStateProperties(object):
     All = RoleConfig | HostingEnv | SharedConfig | ExtensionsGoalState | Certificates | RemoteAccessInfo
 
 
-class GoalStateInconsistentError(ProtocolError):
```

Comment:
Instead of checking the tenant cert at the GoalState level and raising an exception when there is a mismatch, now the callers of GoalState that are actually interested in this situation do the check and handle the error. The caller, in this case the main loop of the extension handler, can do a better job handling and reporting the error.